crypto/fipsmodule/rand/entropy_pool.c

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "internal.h"
+#include "../../internal.h"
+
+#if defined(AWSLC_FIPS)
+
+// entropy_pool_validate_static_assumptions validates static assumptions on the
+// entropy pool |entropy_pool|.
+static int entropy_pool_validate_static_assumptions(
+  struct entropy_pool *entropy_pool) {
+  // By transitivity, |valid_available| and |index_read| are verified to be at
+  // most |ENTROPY_POOL_SIZE|.
+  if (entropy_pool != NULL &&
+    entropy_pool->capacity <= ENTROPY_POOL_SIZE &&
+    entropy_pool->valid_available <= entropy_pool->capacity &&
+    entropy_pool->index_read <= entropy_pool->capacity) {
+    return 1;
+  }
+  return 0;
+}
+
+// entropy_pool_ensure_can_satisfy returns 1 if the entropy pool |entropy_pool|
+// contains enough entropy to satisfy a get request of size |get_size|.
+// Returns 0 otherwise.
+static int entropy_pool_ensure_can_satisfy(struct entropy_pool *entropy_pool,
```

Comment:
Should this abort if `entropy_pool == NULL`?. I know you gate some of the functions already, but not all of them.